web/src/hooks/api/status.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) [2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import { useEffect } from "react";
+import { useSuspenseQuery, useQueryClient } from "@tanstack/react-query";
+import { useInstallerClient } from "~/context/installer";
+import { getStatus } from "~/api";
+import { Status, Progress } from "~/api/status";
+import { isEqual, remove, replaceOrAppend } from "radashi";
+
+const statusQuery = {
+  queryKey: ["status"],
+  queryFn: getStatus,
+};
+
+function useStatus(): Status | null {
+  return useSuspenseQuery(statusQuery)?.data;
+}
+
+// FIXME: Borrowed from radashi 12.7. Simply import it after updating the dependency.
+function isArrayEqual<T>(array1: T[], array2: T[]): boolean {
```

Comment:
I think I already see it in @dgdavid PR. Can it be just on single place?

Reply:
Yes, it is only moved here as part of solving a git conflict. Note that this function is going to be drop once we update the radashi dependency.